storage/src/adb/operation.rs

```diff
@@ -154,73 +156,51 @@ impl<K: Array, V: Array> Write for Operation<K, V> {
 
 impl<K: Array, V: Array> Read for Operation<K, V> {
     fn read_cfg(buf: &mut impl Buf, _: &()) -> Result<Self, CodecError> {
-        let mut value = vec![0u8; Self::SIZE];
-        buf.copy_to_slice(&mut value);
-        Self::try_from(&value).map_err(|e: Error<K, V>| CodecError::Wrapped("Operation", e.into()))
-    }
-}
-
-impl<K: Array, V: Array> FixedSize for Operation<K, V> {
-    const SIZE: usize = K::SIZE + 1 + V::SIZE;
-}
-
-impl<K: Array, V: Array> Array for Operation<K, V> {
-    type Error = Error<K, V>;
-}
-
-impl<K: Array, V: Array> TryFrom<&[u8]> for Operation<K, V> {
-    type Error = Error<K, V>;
-
-    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
-        if value.len() != Self::SIZE {
-            return Err(Error::InvalidLength);
-        }
-
-        let _ = K::try_from(&value[1..K::SIZE + 1]).map_err(|e| Error::InvalidKey(e))?;
-
-        match value[0] {
+        // Create a vector of the required size and copy the data into it.
+        at_least(buf, Self::SIZE)?;
```

Comment:
This section of code feels like a good opportunity for a helper (for a vec with externally defined size)?

Reply:
Yes we have the same issue with [Poly](https://github.com/commonwarexyz/monorepo/blob/f6c7eff445c26a6d569550f7f614336791e5a93c/cryptography/src/bls12381/primitives/poly.rs#L274)

Kind of strange to finagle since we have an implementation for `Vec<u8>` and for `[u8; N]` already, but not for `Vec<T>` of an exact size

Reply:
https://github.com/commonwarexyz/monorepo/issues/788